app/helpers/application_helper.rb

```diff
@@ -1,19 +1,45 @@
 module ApplicationHelper
-  def next_action_button(pull_request, options = {})
-    next_action = TicketActions.for(pull_request).first
+  def next_action_button(pull_request, user: nil, **options)
+    next_action = TicketActions.next_action_for(pull_request, user: user)
     return '' if next_action.blank?
 
-    btn_type =
+    btn_class =
       case next_action.type
-      when :positive then 'success'
-      when :warning then 'warning'
-      when :negative then 'danger'
-      else 'default'
+      when :positive then 'btn-success'
+      when :warning, :caution then 'btn-caution'
+      when :negative then 'btn-danger'
+      else 'btn-default'
       end
 
-    options[:class] = ["btn btn-#{btn_type}", *options[:class]]
-    options[:target] = :_blank
+    options[:class] = ['btn', btn_class, *options[:class]]
 
-    link_to next_action.text, next_action.url, options
+    if next_action.urls.many?
+      options[:type] = 'button'
+      options[:class] << 'dropdown-toggle'
+      options.deep_merge!(data: { toggle: 'dropdown' })
+
+      content_tag(:div, class: 'dropdown') do
+        button_tag(options) do
+          h(next_action.text) + '&nbsp;'.html_safe + content_tag(:span, '', class: 'caret')
+        end + content_tag(:ul, class: 'dropdown-menu dropdown-menu-right') do
```

Comment:
wonder if we can avoid having one `end` and the next `do` on the same line?

Reply:
Yeah, this is pretty hairy. Will split it up a bit.